### PR TITLE
Bump Envoy to v1.20.1

### PR DESCRIPTION
**Pull request by sunjayBhatia**
See release notes:
https://www.envoyproxy.io/docs/envoy/v1.20.0/version_history/current

Also bumps go-control-plane to latest

Deprecation of existing headermatch usage will need to be handled in the release after this is
merged. The "string match" header matching method is preferred but was
added in 1.20.1 and it was not present in 1.19.x, we will need to
release a version with Envoy 1.20.x compatibility and the subsequent
release will need to move to the "string match" method (rather than the
individual regex/exact/etc. methods.

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
Not sure if we want to merge for 1.19 or not cc @projectcontour/maintainers

File: internal/xds/v3/snapshotter.go
Comment (skriss) [inline internal/xds/v3/snapshotter.go:49]: Looks like we could probably update this method signature to take in a `map[envoy_resource_v3.Type][]envoy_types.Resource`, and update the caller to pass that in. Then could use `resources` directly here.

Comment (sunjayBhatia) [reply internal/xds/v3/snapshotter.go:49]: ah yeah let me rework that, theres some room for improvement and de-duplication

**Review by skriss — APPROVED** [review]
:+1: I don't see any reason not to ship it in Contour 1.19.

**Comment by sunjayBhatia** [comment]
added `do not merge` since this is a bit of a curious one, tests seem to be failing in CI but not locally "on my machine"

**Comment by stevesloka** [comment]
I tried running on my machine and seems like some tests are timing out. 🤔

**Comment by sunjayBhatia** [comment]
> > I tried running on my machine and seems like some tests are timing out. 🤔
> 
> yeah latest CI runs ~140s, my local machine ~975 and CI ~882, some issues here for sure

just switching back to the older Envoy, I get ~140s locally, drastic change from the 1.20.0 image

**Comment by sunjayBhatia** [comment]
See https://github.com/envoyproxy/envoy/issues/18616

**Comment by sunjayBhatia** [comment]
testing with image `docker.io/envoyproxy/envoy-dev:6f0ecc5ae16f3c39937dd13a3bc0327177a0e754` (and removed http client timeouts in tests) which has the fix from https://github.com/envoyproxy/envoy/pull/18686

**Comment by wrowe** [comment]
@sunjayBhatia could you fix the PR title? Thx!

**Review by youngnick — APPROVED** [review]

**Review by skriss — APPROVED** [review]
🎉

**Pull request merged**
